src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -203,21 +204,40 @@ std::shared_ptr<JitOperatorWrapper> JitAwareLQPTranslator::_try_translate_sub_pl
   } else {
     if (use_limit) jit_operator->add_jit_operator(std::make_shared<JitLimit>());
 
-    // Add a compute operator for each computed output column (i.e., a column that is not from a stored table).
-    auto write_table = std::make_shared<JitWriteTuples>();
-    for (const auto& column_expression : node->column_expressions()) {
-      const auto jit_expression =
-          _try_translate_expression_to_jit_expression(*column_expression, *read_tuples, input_node);
-      if (!jit_expression) return nullptr;
-      // If the JitExpression is of type JitExpressionType::Column, there is no need to add a compute node, since it
-      // would not compute anything anyway
-      if (jit_expression->expression_type() != JitExpressionType::Column) {
-        jit_operator->add_jit_operator(std::make_shared<JitCompute>(jit_expression));
+    // Check if the output has to be materialized
```

Comment:
Yes, that's what I would expect a variable `output_must_be_materialized` to do. A better comment would explain the reasoning behind that.